app/App.tsx

```diff
@@ -22,6 +22,36 @@ const App: React.FC<AppProps> = ({ children }) => {
     const experienceContext = useExperienceContext();
     const loadingState = isClient ? experienceContext.loadingState : '';
     const [displayLoading, setDisplayLoading] = useState<boolean>(true);
+    const [isPortraitMode, setIsPortraitMode] = useState<boolean>(false);
+    const isPortraitModeRef = useRef<boolean>(isPortraitMode);
+
+    // Function to update orientation mode
+    useEffect(() => {
+      setIsClient(true);
+      setIsPortraitMode(window.innerHeight > window.innerWidth);
+      isPortraitModeRef.current = window.innerHeight > window.innerWidth;
+  
+      const updateOrientationMode = () => {
+        setIsPortraitMode(window.innerHeight > window.innerWidth);
+        isPortraitModeRef.current = window.innerHeight > window.innerWidth;
+      };
+  
+      const handleOrientationChange = () => {
+        updateOrientationMode();
+      };
+  
+      const handleWindowResize = () => {
+        updateOrientationMode();
+      };
+  
+      window.addEventListener('orientationchange', handleOrientationChange);
+      window.addEventListener('resize', handleWindowResize);
```

Comment:
Why not just:

```
const updateOrientationMode = () => {
  setIsPortraitMode(window.innerHeight > window.innerWidth);
  isPortraitModeRef.current = window.innerHeight > window.innerWidth;
};

window.addEventListener('orientationchange', updateOrientationMode);
window.addEventListener('resize', updateOrientationMode);
```

Since `handleOrientationChange` and `handleWindowResize` are just calling `updateOrientationMode`.